libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/TelemetryInitializerTests.java

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.applicationinsights;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.bot.applicationinsights.core.TelemetryInitializerMiddleware;
+import com.microsoft.bot.builder.BotTelemetryClient;
+import com.microsoft.bot.builder.TelemetryLoggerMiddleware;
+import com.microsoft.bot.builder.adapters.TestAdapter;
+import com.microsoft.bot.builder.adapters.TestFlow;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.ActivityTypes;
+
+public class TelemetryInitializerTests {
+
+    @Test
+    public void telemetryInitializerStoresActivity() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+
+        /**
+         * TODO: Check Typescript implementation
+         */
+    }
+
+    @Test
+    public void telemetryInitializerMiddlewareLogActivitiesWhenEnabled() {
+    	TelemetryClient mockTelemetryClient = Mockito.mock(TelemetryClient.class);
+
+        TelemetryLoggerMiddleware telemetryLoggerMiddleware = new TelemetryLoggerMiddleware((BotTelemetryClient) mockTelemetryClient, false);
+
+        TestAdapter testAdapter = new TestAdapter()
+        		.use(new TelemetryInitializerMiddleware(telemetryLoggerMiddleware, true));
+
+        TestFlow testFlow = new TestFlow(testAdapter, (turnContext) -> {
+            Activity typingActivity = new Activity() {
+            {
+            	setType(ActivityTypes.TYPING);
+            	setRelatesTo(turnContext.getActivity().getRelatesTo());
+            }};
+            turnContext.sendActivity(typingActivity).join();
+            try {
+                TimeUnit.SECONDS.sleep(5);
+            } catch (InterruptedException e) {
+                // Empty error
+            }
+            turnContext.sendActivity(String.format("echo:%s", turnContext.getActivity().getText())).join();
+            return CompletableFuture.completedFuture(null);
+        });
+
+        testFlow.send("foo")
+		        .assertReply(activity -> {
+		            Assert.assertTrue(activity.isType(ActivityTypes.TYPING));
+		        })
+		        .assertReply("echo:foo");
+
+        verify(mockTelemetryClient, times(1));
```

Comment:
This is not okay, check [this](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerTests.java#L1457), it should something similar

Reply:
yes, I've seen that, but do we need to capture those values?

Reply:
After reviewing this with Victor, it's okay as it's right now, don't change anything here